docs/process.md

```diff
@@ -0,0 +1,100 @@
+## Process
+
+Process models a set of manufacturing operations to be executed
+(i.e. [Tasks](task.md)).
+Process references Process Definition that specifies how to perform the work in detail.
+
+Process is not executed directly,
+but its constituent Tasks are executed by concrete agents.
+
+### Example #1 - Creating Process   
+
+A party requesting unloading supplies creates Process
+that specifies an existing Process Definition and some work parameters:
+
+```json
+{
+    "id": "urn:ngsi-ld:Process:company-xyz:nc847rc8347nc3",
+    "type": "Process",
+    "isDefinedBy": {
+        "type": "Relationship",
+        "object": "urn:ngsi-ld:ProcessDefinition:company-xyz:unloading-supplies"
+    },
+    "workParameters": {
+        "type": "Property",
+        "value": {
+            "truckId": 838
+        }
+    },
+    "status": {
+        "type": "Property",
+        "value": "pending",
+        "observedAt": "2020-12-01T11:23:19Z"
+    },
+    "@context": [
+        "https://smartdatamodels.org/context.jsonld",
+        "https://raw.githubusercontent.com/shop4cf/data-models/master/docs/shop4cfcontext.jsonld"
+    ]
+}
+```
+
+Required attributes are: `id`, `type`, `isDefinedBy`, `status`.
+
+Further notes on the attributes:
+- `isDefinedBy` references a Process Definition (which does not rather live in Context Broker)
+- `workParameters` are use-case specific, as in [Task](task.md)
+- `status` represents the current execution status, as in [Task](task.md)
```

Comment:
I have one addition coming from the use case of SAG. The Status should include also:
`approved` , `notapproved` . This will be a status updated by RA whenever a Process is published to confirm if the safety for that process is ok or not.

Reply:
Good point regarding approval but it might be a bit tricky (or at least it seems difficult for me to grasp).
Since RA tool runs in design time (offline), then the process (instance) has not yet started. So, the tool "runs/simulates" the execution on the process model (i.e., definition) and not on the actual running instance of it. So, what shall then be labeled as "approved" (if it doesn't exist yet)? The idea I guess is to prevent sth from being started as it would be considered not sage.

Also, I'm not sure I've understood the use of the RA tool. If let's say it finds that running a process with a KUKA robot on one task and an IWA robot on another task is not safe, what happens then? They re-run their tool to find a combination that is safe? Maybe their results should somehow be saved somewhere (out of CB I think), and then during runtime the resource allocation mechanism (on MPMS I guess) checks these results and performs the right/allowed allocations.
So, for me I see that the "approval" tag should be stored before the actual run of an instance (by the way, is the approval on the level of the whole process or on individual tasks?).
Unless we create "non-actual" (or dummy or another name that shows that is not the actual execution) process instances when the RA tools runs. Then in that case, these dummy process instances can be indeed marked as (non)approved. But then, should MPMS check these dummy process instances before the actual resource allocation?
Big discussion, and I think it's hard for me to grasp things if I don't know the full lifecycle/goals of the RA tool.

Reply:
Also, I think the "status" in process entity should not (doesn't make) sense to have all the values from task status.
A process instance might have tasks that have been finished, tasks that are active (e.g. some of them in progress, others assigned, others paused which do not affect the continuation of a parallel workflow), and tasks that are to be executed at a later point. So, how should we call this process instance? In progress (makes sense)?
And then there should be some logic, e.g. if a paused task blocks everything else from running, then the process is also paused.
So, the state-diagram of the values of the process statuses to me should be different to the one for tasks (and not all values are relevant)

Reply:
@kontrag I have seen this just now. Let's briefly discuss in person tomorrow

Reply:
@kontrag, re. the values of `Process.status`: I wanted to "reuse" the already defined `Tasks.status` attribute for simplicity. I would see the mapping between Process's and related Tasks' statuses as follows:
  - `pending` - execution of this Process is requested, but the component (the agent) responsible for its execution is not yet aware. If the related Tasks exist, they are also pending.
  - `assigned` - execution of this Process is requested, and the component (the agent) responsible for its execution has accepted it. If the related Tasks exist, they are pending or assigned.
  - `inProgress` - actual execution is in progress. At least one Task is inProgress. None has failed. Some may be completed.
  - `completed` - execution has completed. All Tasks are completed.
  - `paused` - execution has been in progress but is now temporarily paused. At least one Task is paused. None has failed. None is inProgress.
  - `failed` - execution has permanently failed. At least one Task is failed.

A concrete scenario does not need to use/implement all the status values. But in the model, we keep them all to support a wide range of scenarios.

Does it make sense?

(re. the new status values for RA, I will send an email and we should discuss in a call)

Reply:
We agreed to keep the proposed status values but without providing a concrete logic/rules as in the comment above.